sw/device/boot_rom2/ext_format.c

```diff
@@ -0,0 +1,130 @@
+// Cn:opyright lowRISC contributors.
+// Licensed under the Apache License, Version 2.0, see LICENSE for details.
+// SPDX-License-Identifier: Apache-2.0
+
+#include "sw/device/boot_rom2/ext_format.h"
+
+#include "sw/device/lib/base/memory.h"
+#include "sw/device/lib/runtime/check.h"
+
+static const uint32_t kExtMagic = 0x5245545f;
+
+/**
+ * Convenience struct for pulling out the fixed part of the executable header.
+ */
+typedef struct fixed_header {
+  uint32_t magic, version, payload_len, entrypoint;
+} fixed_header_t;
+
+static size_t metadata_len_from_version(ext_format_version_t version) {
+  switch (version) {
+    case kExtFormatVersion0:
+      return 0;
+    default:
+      return 0;
+  }
+}
+
+/**
+ * A struct containing all non-version-specific information parsed out of a
+ * binary.
+ *
+ * This struct does not include signature information, which is already verified
+ * at this point.
+ */
+typedef struct common_parse_result {
+  /**
+   * Absolute address of the start of the metadata region.
+   */
+  const uint8_t *metadata;
+  /**
+   * Length of the metadata region.
+   */
+  size_t metadata_len;
+  /**
+   * Absolute address of the start of the payload region.
+   */
+  const uint8_t *payload;
+  /**
+   * Length of the payload region.
+   */
+  size_t payload_len;
+  /**
+   * Absolute address of the next stage entrypoint.
```

Comment:
nit: Is this the entry point of the code partition being verified?  Why not say something more explicit like "payload entrypoint"?

Reply:
Yes. I thought that that would strongly implicit, given the parallel with loading an ELF file? I can change it if you still disagree.